### ISSUE TITLE
[BUG] Cone mesh generion might have incorrect normals

### ISSUE THREAD

**Issue by crsaracco** [amethyst/amethyst#1922]
## Description

I noticed some very weird lighting issues when I generated a cone by doing:

```rust
let cone_mesh = world.exec(|loader: AssetLoaderSystemData<'_, Mesh>| {
    loader.load_from_data(
        Shape::Cone(100)
            .generate::<(Vec<Position>, Vec<Normal>, Vec<Tangent>, Vec<TexCoord>)>(None)
            .into(),
        (),
    )
});
```

I made a full working example here: https://gist.github.com/crsaracco/024c5b2801a8bedea35cb58e55d186e2

And here's a screenshot:

![broken](https://i.imgur.com/jJVw2U7.png)

It was mentioned on discord that either the normal generation or the triangle vertex ordering might be wrong, so I tried editing these two z-axis normals in `gfx-rs/genmesh` ([[1]](https://github.com/gfx-rs/genmesh/blob/master/src/cone.rs#L48), [[2]](https://github.com/gfx-rs/genmesh/blob/master/src/cone.rs#L59)) from `-FRAC_1_SQRT_2` to `FRAC_1_SQRT_2` (I.E. I flipped the sign), and it seemingly fixed it:

![fixed](https://i.imgur.com/YlWsa18.png)

I'm not sure if the bug is in the Cone generation in `gfx-rs/genmesh`, or if it's further up the call stack.

## Your Environment

 - **Operating System**: Arch linux
 - **Version of Amethyst**: 0.12.0

**Comment by crsaracco** [comment]
For triaging (I don't think I can edit the tags):

> Can you file a ticket with these screenshots and that statement and set it to team: rendering assigned to me?
> I'll see if it's genmesh or us and upstream as needed :)
 - @jaynus

**Comment by crsaracco** [comment]
Looks like this has been resolved between 0.12.0 and 0.13.2.
